File: 18-Fibonacci.cpp
// 18. Write a program to display Fibonacci series (i) using recursion, (ii) using iteration

#include <iostream>
using namespace std;

// using recursion
int fib(int x)
{
    if (x == 1)
        return 0;
    if (x == 2)
        return 1;
    else
        return (fib(x - 1) + fib(x - 2));
}

int main()
{
    // using iteration
    int i, a = 0, b = 1, c;
    cout << a << " " << b << " ";

    for (i = 1; i < 10; i++)
    {
        c = a + b;
        a = b;
        b = c;
        cout << c << " ";
    }
    cout << endl;

    for (i = 1; i < 12; i++)
        cout << fib(i) << " ";
    cout << endl;
    return 0;
}

File: 4th_Semester/PHP/Practicals/18_fibonacci.php
<!-- 18. Write a PHP program to print Fibonacci series using recursion. -->

<?php
$FILENAME = pathinfo(__FILE__, PATHINFO_FILENAME) . "." . pathinfo(__FILE__, PATHINFO_EXTENSION);
echo "<header><h1>FileName: " . $FILENAME . "</h1></header></br></br>";
?>

<head>
    <link rel="stylesheet" href="forPhp.css">
</head>

<body>
    <div class="form-container">
        <form action="" method="post">
            How many number of Fibonacci do you wanna see?
            </br></br>
            <input type="number" min="1" name="limit"><br><br>
            <input type="submit" value="Submit">
        </form>
    </div>
</body>

</html>

<?php
if (!isset($_POST['limit'])) return;

$limit = $_POST['limit'];

if ($limit == null) return;

function fib($limit)
{
    if ($limit == 0) return 0;
    else if ($limit == 1) return 1;
    else return fib($limit - 1) + fib($limit - 2);
}

echo "<div class='result'>";
for ($i = 0; $i < $limit; $i++) {
    echo fib($i) . " ";
}
echo  "</div>";
?>